RecoLocalTracker/SiPixelRecHits/plugins/PixelCPETemplateRecoESProducer.cc

```diff
@@ -30,35 +30,41 @@ class PixelCPETemplateRecoESProducer : public edm::ESProducer {
   edm::ESGetToken<SiPixelTemplateDBObject, SiPixelTemplateDBObjectESProducerRcd> templateDBobjectToken_;
 
   edm::ParameterSet pset_;
-  bool DoLorentz_;
+  bool doLorentzFromAlignment_;
+  bool useLAFromDB_;
 };
 
 using namespace edm;
 
 PixelCPETemplateRecoESProducer::PixelCPETemplateRecoESProducer(const edm::ParameterSet& p) {
   std::string myname = p.getParameter<std::string>("ComponentName");
 
-  //DoLorentz_ = p.getParameter<bool>("DoLorentz"); // True when LA from alignment is used
-  DoLorentz_ = p.getParameter<bool>("DoLorentz");
+  useLAFromDB_ = p.getParameter<bool>("useLAFromDB");
+  doLorentzFromAlignment_ = p.getParameter<bool>("doLorentzFromAlignment");
 
   pset_ = p;
   auto c = setWhatProduced(this, myname);
   magfieldToken_ = c.consumes();
   pDDToken_ = c.consumes();
   hTTToken_ = c.consumes();
   templateDBobjectToken_ = c.consumes();
-  if (DoLorentz_) {
+
+  char const* laLabel = "";  // standard LA, from calibration, label=""
+  lorentzAngleToken_ = c.consumes(edm::ESInputTag("", laLabel));
+
+  if (doLorentzFromAlignment_) {
     lorentzAngleToken_ = c.consumes(edm::ESInputTag("", "fromAlignment"));
```

Comment:
(copied from PixelCPEClusterRepairESProducer.cc)

```suggestion
  if (useLAFromDB_ || doLorentzFromAlignment_) {
    char const* laLabel = doLorentzFromAlignment_ ? "fromAlignment" : "";
    lorentzAngleToken_ = c.consumes(edm::ESInputTag("", laLabel));
```

this is matching the actual use case in `::produce` (only one payload is actually consumed; the current implementation can declare two different consumes). However I'm not sure what your actual goal is for the case with both flags set to true.

Reply:
Ok I updated to your suggestion